src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -208,7 +213,11 @@ private Query parseQuery(String type, XContentParser parser) {
             // Query parsing can't introduce new fields in mappings (which happens when registering a percolator query),
             // because field type can't be inferred from queries (like document do) so the best option here is to disallow
             // the usage of unmapped fields in percolator queries to avoid unexpected behaviour
+            //
+            // For backward compatibility, query can contain unmapped fields only if index.percolator.map_unmapped_fields_as_string
+            // is set to true
```

Comment:
I think this bwc comment can be removed? The `index.percolator.map_unmapped_fields_as_string` setting is there just to facilitate in the case a field is unmapped and then automatically creating a default string field.